net/dns/resolver/forwarder.go

```diff
@@ -317,11 +338,24 @@ func (f *forwarder) getDoHClient(ip netaddr.IP) (urlBase string, c *http.Client,
 				if !strings.HasPrefix(netw, "tcp") {
 					return nil, fmt.Errorf("unexpected network %q", netw)
 				}
-				c, err := nsDialer.DialContext(ctx, "tcp", net.JoinHostPort(ip.String(), "443"))
+				toDial := host
+				if _, err := netaddr.ParseIP(toDial); err != nil && len(bootstrapRes) > 0 {
+					// Need to resolve host name, and this is
```

Comment:
I don't want to pay this resolution cost for our current set of knownDoH servers because they're all unnecessary: they have fixed IP addresses.

Only for OpenDNS etc do we need to do DNS lookups for DoH bootstrapped.

Can we do this in a later version? This is adding a new feature (and a new unnecessary cost in some cases) to a PR that should be just about changing types.

Reply:
It doesn't execute for knownDoH because getKnownDoHClient passes in an empty `bootstrapRes` slice so this code never runs.

The idea is we only do the resolution here if we can do a better job than DialContext (which is only if we have a bootstrapRes value.)

I'll look into pulling it out.

Reply:
Well, even when len(bootstrapRes) == 0, aren't you still doing an implicit DNS lookup because `toDial` is the hostname?